### PR TITLE
DCES-480 Flyway

### PR DESCRIPTION
**Pull request by LAA-Dave-Edwards**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Initial Flyway commit.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: dces-drc-integration/build.gradle [hunk 1]
```diff
@@ -121,6 +125,7 @@ dependencies {
 	testImplementation 'io.projectreactor:reactor-test'
 	testImplementation "org.wiremock:wiremock-standalone:$versions.wiremockVersion"
 	testImplementation "com.squareup.okhttp3:mockwebserver:$versions.mockwebserverVersion"
+	testImplementation "org.flywaydb:flyway-core:6.4.3"
```
Comment (aminur-rouf) [inline dces-drc-integration/build.gradle:128]: is there a reason why the testImplementaion version is different to  implementation "org.flywaydb:flyway-core:$versions.flywayVersion" which is 10.20.0 

Does it matter?

Comment (LAA-Dave-Edwards) [reply dces-drc-integration/build.gradle]: Nope, that's my mistake. Updated to latest as well.

File: dces-drc-integration/src/main/resources/db/migration/V1_0_2__base_data.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO record_type VALUES ('Contribution'), ('Fdc') ON CONFLICT DO NOTHING;
```
[the remark this answers is not in the document]
Comment (LAA-Dave-Edwards) [reply dces-drc-integration/src/main/resources/db/migration/V1_0_2__base_data.sql:1]: The naming convention for flyway from all I've seen is:
<Prefix><Version>__<Description>.sql

The version is needed in order to order the scripts. 
In this case: 1.0.2

File: dces-drc-integration/src/test/resources/application.yaml [hunk 3]
```diff
@@ -36,6 +36,9 @@ spring:
     database-platform: org.hibernate.dialect.PostgreSQLDialect
     show-sql: false
     generate-ddl: false
+  flyway:
+    baselineOnMigrate: false
+    enabled: false
```
[the remark this answers is not in the document]
Comment (LAA-Dave-Edwards) [reply dces-drc-integration/src/test/resources/application.yaml:41]: Nope.

**Review by aminur-rouf — APPROVED** [review]

**Pull request merged**